src/controllers/aer_controller.hpp

```diff
@@ -189,6 +187,12 @@ class Controller {
                           const Method method, bool cache_block,
                           ExperimentResult &result) const;
 
+  template <class State_t>
+  void run_batched_circuits_helper(const std::vector<Circuit> &circs, const Noise::NoiseModel &noise,
+                          const json_t &config, 
+                          const Method method, 
+                          Result &result) const;
+
```

Comment:
`run_batched_circuits()` and `run_batched_circuits_helper()` are duplicated.  We can integrate them with `run_circuit()` and two `run_circuit_helper()` with templating two arguments: `Simulatable_t` and `Result_t`. `shots` and `seed` can be retrieved from each `circuit`. Though a flag for multi-cunk is not necessary for batched execution, setting a dummy flag (as `false`) does not produce any confusion, I believe.
```
  template <class Simulatable_t, class Result_t>
  void run_circuit(const Simulatable_t &circ, const Noise::NoiseModel &noise,
                   const Method method,const json_t &config, Result_t &result, bool multi_chunk) const;

  template <class State_t>
  void run_circuit_helper(const Circuit &circ, const Noise::NoiseModel &noise,
                          const json_t &config, const Method method, const bool cache_block,
                          ExperimentResult &result) const;

  template <class State_t>
  void run_circuit_helper(const std::vector<Circuit> &circs, const Noise::NoiseModel &noise,
                          const json_t &config, const Method method, const bool cache_block,
                          Result &result) const;
```

Reply:
integrated these functions